src/storage/admin/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,97 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "kvstore/LogEncoder.h"
+#include "storage/StorageFlags.h"
+#include "storage/admin/RebuildEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace storage {
+
+void RebuildEdgeIndexProcessor::process(const cpp2::RebuildIndexRequest& req) {
+    CHECK_NOTNULL(kvstore_);
+    auto space = req.get_space_id();
+    auto parts = req.get_parts();
+    callingNum_ = parts.size();
+    auto indexID = req.get_index_id();
+    auto itemRet = indexMan_->getEdgeIndex(space, indexID);
+    if (!itemRet.ok()) {
+        cpp2::ResultCode thriftRet;
+        thriftRet.set_code(cpp2::ErrorCode::E_INDEX_NOT_FOUND);
+        codes_.emplace_back(thriftRet);
+        onFinished();
+        return;
+    }
+
+    auto item = itemRet.value();
+    auto edgeType = item->get_schema_id().get_edge_type();
+    LOG(INFO) << "Rebuild Edge Index Space " << space << " Edge Type " << edgeType
+              << " Edge Index " << indexID;
+
+    for (PartitionID part : parts) {
+        if (req.get_is_offline()) {
+            std::unique_ptr<kvstore::KVIterator> iter;
+            auto prefix = NebulaKeyUtils::prefix(part);
+            auto ret = kvstore_->prefix(space, part, prefix, &iter);
+            if (ret != kvstore::ResultCode::SUCCEEDED) {
+                LOG(ERROR) << "Processing Part " << part << " Failed";
+                this->pushResultCode(to(ret), part);
+                onFinished();
+                return;
+            }
+
+            std::vector<kvstore::KV> data;
+            data.reserve(FLAGS_rebuild_index_batch_size);
+            int32_t batchSize = 0;
+            VertexID currentSrcVertex = -1;
+            VertexID currentDstVertex = -1;
+            while (iter && iter->valid()) {
+                if (batchSize >= FLAGS_rebuild_index_batch_size) {
+                    callingNum_ += 1;
```

Comment:
The way you calculate callingNum_ is not correct.   You should set the callingNum_ before calling doPut,  and don't update it again.